### PR TITLE
add SCROUT functionality

### PR DESCRIPTION
**Pull request by mdasoh**

### CODE REVIEW

**Review by mdasoh — COMMENTED** [review]
gwdata.asm had to move to the front of the line because it states the data segment must begin there.

**Comment by dmsc** [comment]
See #6 for another implementation of the same.

**Comment by tkchia** [comment]
Hello @mdasoh, hello @dmsc,

It seems to me that we do need to use function `9` rather than `0AH`, so that the character colour will be set correctly.  Also, yes, it is kind of puzzling that the released source code does not include a stack segment.

Thank you!

**Comment by sgros** [comment]
In SCNDRV.ASM there is the following line:

DOESC:  CALL    SCROUT          ; Returns carry set if continuation of escape sequence

This probably means that SCROUT has to clear carry on exit when it prints "odrinary" character.

**Comment by dspinellis** [comment]
Given that the code was missing a stack, I added a stack specification in the link script. Can you please remove it from your commit and rebase?

File: OEM.ASM [hunk 1]
```diff
@@ -312,4 +326,46 @@ MAPSUP:			;Dummy function
 
 CSEG	ENDS
 	END
-
+
+SSEG SEGMENT stack 'STACKSG'
+ASSUME SS:SSEG
+	DB	0,0,0,0,0,0,0,0, 0,0,0,0,0,0,0,0
```
Comment (dspinellis) [inline OEM.ASM]: We can write this with 
```
REPT	100
	DB	0
ENDM 
```
But I added a link directive to achieve the same.

File: OEM.ASM [hunk 2]
```diff
@@ -312,4 +326,46 @@ MAPSUP:			;Dummy function
 
 CSEG	ENDS
 	END
-
+
```
Comment (mdasoh) [inline OEM.ASM]: Not too sure, just hope this does the trick.
[OEM.ASM.txt](https://github.com/dspinellis/GW-BASIC/files/4691929/OEM.ASM.txt)

Comment (mdasoh) [inline OEM.ASM]: Guess that didn't work either.

**Pull request merged**